### DISCUSSION TITLE
Is i3 right for me?

### DISCUSSION THREAD

**Discussion by 127Gar** [i3/i3#4561] [Q&A]
Hello all,
If this isn't appropriate, please inform me so that I can change my behaviour.

I am a long-term user of Ubuntu, with probably upper basic level knowledge. I'm comfortable and competent in working in following online instructions for the terminal, and I have limited knowledge independently. I want to challenge myself and improve my computer skills, and I really like the idea of a tiling window manager. I am very fast to learn and willing. I just got a new laptop on sale, and I plan on trying Fedora on it, and otherwise experimenting. If I like i3 on it, I plan on trying it on my current Ubuntu 21.04 ThinkPad.

Do you think i3 would too advanced or difficult for me? And if you do think it makes sense, do you think I should try it on Fedora first or my main Ubuntu machine?

Thanks for the help!

EDIT: I forgot to mention that I mostly word process with LibreOffice, read PDF documents, have a notepad programme open, use Zoom for school, Anki for studying foreign languages, and use Firefox. I'm starting to learn Python with Mu and graphics with Inkscape.

**Comment by ghost** [comment]
How about installing in a VM and trying different customized versions? Vanilla i3 is solid as a rock, however it has little aesthetic merit, truth be told. That's where ricing and all the fun begins. I started with Regolith which I simply didn't like OOB, then found Erik Dubois' i3-gaps installation script for Mint, then switched to ArcoLinux i3, and now my i3 is so different from default Arco I can't use Meld to compare configs. :)

I've been on i3 for over two years and it's going to remain my daily driver into the foreseeable future. It simply works.

Everyone has a different path. Make yours count.

**Comment by Jonta** [comment]
- Install
- Log out
- Log in with i3
- Go through the setup
- Put different windows in different places
- Resize them
- Launch programs with dmenu
- Move windows from one workspace to another
- Look at the config file
- Install i3-blocks and find how surprisingly easy it is to add and modify stuff that's in i3-bar
- Don't be too hard on yourself for having waited that long to try it

**Comment by MillingtonAllen** [comment]
To answer your question directly, yes i3 is for you.  If you try i3, you will probably like it, but there's a bit of a learning curve.  You'll learn it within a couple weeks, if not sooner.  The containerization will be frustrating until you get the hang of it.

Before installing it, print up the keyboard shortcut guide:
https://i3wm.org/docs/userguide.html

When you're comfortable with the basics, dabble with changing workspace names to suit your workflow in the ~/.config/i3/config file.  I have it as
1: Web
2: Utility
3: Media
5: Lecture videos
6,7,8,9,0 for different classes

Be aware, I believe workspaces are sorted alphabetically (unless there's a way to re-order that I'm not aware of), so keeping the initial numbering prefix helps keep the ordering right.

After that, try adding your own exec shortcuts.  Like which programs you want to start on startup, or assigning keyboard shortcuts to execute files.  I had $mod + Shift + e for Eclipse, though now I've assigned it to start on boot.  

Dabble with colors when you're comfortable.  It's fairly easy, don't let editing text files scare you off.  Just make backup copies EACH TIME before changing anything.  If there's any problems in the config file, I believe it defaults to a default config, so it's not like the system will be broken.  If you don't make backup copies, it will be hard to trace what is broken.

When you get into thing starting on boot, you'll want to look into how to have them open in a specific workspace automatically.

Take things one step at a time, and you'll be fine.  Personal preference:  I found i3 works well on top of a Kubuntu install.  Overall I prefer KDE over Gnome.
